x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java

```diff
@@ -906,19 +904,12 @@ private static String typeErrorMessage(boolean includeOrdinal, List<Set<DataType
             "boolean or cartesian_point or datetime or geo_point or numeric or string"
         ),
         Map.entry(
-            Set.of(EsqlDataTypes.GEO_POINT, DataTypes.KEYWORD, DataTypes.LONG, DataTypes.TEXT, DataTypes.UNSIGNED_LONG, DataTypes.NULL),
-            "geo_point or long or string or unsigned_long"
+            Set.of(EsqlDataTypes.GEO_POINT, DataTypes.KEYWORD, DataTypes.LONG, DataTypes.TEXT, DataTypes.NULL),
+            "geo_point or long or string"
         ),
         Map.entry(
-            Set.of(
-                EsqlDataTypes.CARTESIAN_POINT,
-                DataTypes.KEYWORD,
-                DataTypes.LONG,
-                DataTypes.TEXT,
-                DataTypes.UNSIGNED_LONG,
-                DataTypes.NULL
-            ),
-            "cartesian_point or long or string or unsigned_long"
+            Set.of(EsqlDataTypes.CARTESIAN_POINT, DataTypes.KEYWORD, DataTypes.LONG, DataTypes.TEXT, DataTypes.NULL),
+            "cartesian_point or long or string"
```

Comment:
Sorry this code is such a mess. It's trying to be cute and guess error messages but I think it can get a bit annoying to maintain.